util/math.h

```diff
@@ -20,6 +23,26 @@ ASSERT_FEATURE_COMPAT_HEADER();
 
 namespace ROCKSDB_NAMESPACE {
 
+// Fast implementation of extracting the bottom n bits of an integer.
+// To ensure fast implementation, undefined if n bits is full width or more.
+template <typename T>
```

Comment:
Just curious why you decided to make this template not concerned about references as the "BitwiseAnd" template does.

Reply:
That is a legitimate inconsistency between the implementations! It looks like my remove_reference are unnecessary. https://effective-modern-cpp.readthedocs.io/en/latest/chapter1.html https://stackoverflow.com/questions/26631169/why-does-sizeof-a-reference-type-give-you-the-sizeof-the-type

Reply:
But I'll generate a good message if someone uses cref as an explicit type parameter.